packages/flutter_tools/bin/fuchsia_asset_builder.dart

```diff
@@ -0,0 +1,110 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:args/args.dart';
+import 'package:process/process.dart';
+
+import '../lib/src/asset.dart';
+import '../lib/src/base/config.dart';
+import '../lib/src/base/context.dart';
+import '../lib/src/base/file_system.dart';
+import '../lib/src/base/io.dart';
+import '../lib/src/base/logger.dart';
+import '../lib/src/base/os.dart';
+import '../lib/src/base/platform.dart';
+import '../lib/src/cache.dart';
+import '../lib/src/devfs.dart';
+import '../lib/src/disabled_usage.dart';
+import '../lib/src/flx.dart';
+import '../lib/src/globals.dart';
+import '../lib/src/usage.dart';
+
+const String _kOptionPackages = 'packages';
+const String _kOptionWorking = 'working-dir';
+const String _kOptionManifest = 'manifest';
+const String _kOptionAssetManifestOut = 'asset-manifest-out';
+const List<String> _kRequiredOptions = const <String>[
+  _kOptionPackages,
+  _kOptionWorking,
+  _kOptionAssetManifestOut,
+];
+
+Future<Null> main(List<String> args) async {
+  final AppContext executableContext = new AppContext();
+  executableContext.setVariable(Logger, new StdoutLogger());
+  await executableContext.runInZone(() {
+    // Initialize the context with some defaults.
```

Comment:
Ooph - another copy of this bootstrap code.  Can you refactor to share this between here and `fucshia_builder.dart`?

Reply:
Perhaps:

    typedef Future<Null> Runner(List<String> args);

    Future<Null> runInContext(List<String> args, Runner runner) {
      ...
    }

Reply:
Okay, I did something, hopefully it is approximately correct.